lib/util/schema-utils.ts

```diff
@@ -94,16 +94,19 @@ export function looseValue<T, U extends z.ZodTypeDef, V>(
 export function parseJson<
   T = unknown,
   Schema extends z.ZodType<T> = z.ZodType<T>
->(input: string, schema: Schema): z.infer<Schema> {
-  const parsed = JSON.parse(input);
+>(input: unknown, schema: Schema): z.infer<Schema> {
+  let parsed = input;
+  if (typeof input === 'string') {
```

Comment:
shouldn't we throw if it's not a string?

Reply:
If provided non-string value doesn't match the schema, it will throw

Reply:
then the function name is misleading. as we just validate the value against the schema when not string.

Reply:
Schema can transform data, in this case it can be called parsing

Reply:
if you plan to use it for non json string input, then we should rename it. 

if we only use it to parse and validate json strings, then we should validate input to be string and throw if it's not string.